docs/user/installation.rst

```diff
@@ -4,50 +4,239 @@
+<http://deeplearning.net/software/theano/install.html>`_ which cover installing
+the prerequisites for a range of operating systems, or ask for help on `our
+mailing list <https://groups.google.com/d/forum/lasagne-users>`_.
 
-Install from PyPI
-=================
 
-.. note:: Lasagne hasn't been released as of now.  The only way to
-          install it for now is from source.
+Prerequisites
+=============
 
-To install release 0.1 of Lasagne from PyPI, run the following
-commands:
+Python
+------
+
+Lasagne currently requires Python 2.7 or 3.4 to run. Please install Python via
+the package manager of your operating system if they are not included already.
+Use Python 2.7 if you're in doubt; it is still more widespread than Python 3.
+
+C compiler
+----------
+
+Theano requires a working C compiler, and numpy/scipy require a compiler as
+well if you install them via ``pip``. On Linux, the default compiler is usually
+``gcc``, and on Mac OS, it's ``clang``. Again, please install them via the
+package manager of your operating system.
+
+numpy/scipy + BLAS
+------------------
+
+Lasagne requires numpy of version 1.6.2 or above, and Theano also requires
+scipy 0.11 or above. Numpy/scipy rely on a BLAS library to provide fast linear
+algebra routines. They will work fine without one, but a lot slower, so it is
+worth getting this right (but less importantly so if you plan to use a GPU).
+
+If you install numpy and scipy via your operating system's package manager,
+they should link to the BLAS library installed in your system. If you install
+numpy and scipy via ``pip install numpy`` and ``pip install scipy``, make sure
+to have development headers for your BLAS library installed (e.g., the
+``libopenblas-dev`` package on Debian/Ubuntu) while running the installation
+command. Please refer to the `numpy/scipy build instructions
+<http://www.scipy.org/scipylib/building/index.html>`_ if in doubt.
+
+Theano
+------
+
+The version to install depends on the Lasagne version you choose, so this will
+be handled below.
+
+
+Stable Lasagne release
+======================
+
+.. note:: Lasagne hasn't been released yet. Please see the next section for
+    how to install the bleeding-edge version instead.
+
+Lasagne 0.1 requires a more recent version of Theano than the one available
+on PyPI. To install a version that is known to work, run the following command:
 
 .. code-block:: bash
 
   pip install -r https://raw.githubusercontent.com/Lasagne/Lasagne/0.1/requirements.txt
+
+A more recent version will often work as well, but at the time of writing, a
+simple ``pip install Theano`` will give you a version that is too old.
+
+To install release 0.1 of Lasagne from PyPI, run the following command:
+
+.. code-block:: bash
+
   pip install Lasagne==0.1
 
-Install from source
-===================
+If you do not have administrative rights on your machine, add ``--user`` to
+both commands to install into your home directory instead. To upgrade from an
```

Comment:
> For me, everything's fine without virtualenv, mostly because I've 
> never had a package complain about a dependency being too new. So for 
> me, it works well to install the pip version of a module if the one 
> from the Ubuntu repository is too old, and then it automatically 
> takes precedence over the repository version so I don't have to 
> uninstall the latter (possibly leaving Ubuntu package dependencies 
> unresolved).
> 
OK, fine.  If we can get rid of suggesting things should be installed 
with "sudo" (aka administrative rights), then we should be good.  I do 
sincerely think that it can fuck up your computer if you're sudo pip 
installing packages that your operating system depends on; Linux 
distros tend to use a lot of Python packages for core functionality 
these days.  (I realize that Theano and numpy probably don't qualify, 
but people will use this experience to install more common dependencies 
as well.)

As a side note: if you use virtualenv with the `--system-site-packages` 
flag, you can also go on using your Ubuntu's numpy.
> It should be as easy as possible to get from zero to Lasagne.
> 
... without screwing up your OS.  ;-)